pyt/__main__.py

```diff
@@ -39,66 +57,74 @@ def main(command_line_args=sys.argv[1:]):  # noqa: C901
+            nosec_lines = set(
+                lineno for
+                (lineno, line) in enumerate(lines, start=1)
+                if '#nosec' in line or '# nosec' in line
+            )
 
-    if args.ignore_nosec:
-        nosec_lines = set()
-    else:
-        file = open(path, 'r')
-        lines = file.readlines()
-        nosec_lines = set(
-            lineno for
-            (lineno, line) in enumerate(lines, start=1)
-            if '#nosec' in line or '# nosec' in line
-        )
+        if args.project_root:
+            directory = os.path.normpath(args.project_root)
+        else:
+            directory = os.path.dirname(path)
+        project_modules = get_modules(directory)
+        local_modules = get_directory_modules(directory)
+        tree = generate_ast(path)
 
-    if args.project_root:
-        directory = os.path.normpath(args.project_root)
-    else:
-        directory = os.path.dirname(path)
-    project_modules = get_modules(directory)
-    local_modules = get_directory_modules(directory)
+        cfg = make_cfg(
+            tree,
+            project_modules,
+            local_modules,
+            path
+        )
+        cfg_list = [cfg]
 
-    tree = generate_ast(path)
 
-    cfg = make_cfg(
-        tree,
-        project_modules,
-        local_modules,
-        path
-    )
-    cfg_list = [cfg]
-    framework_route_criteria = is_flask_route_function
-    if args.adaptor:
-        if args.adaptor.lower().startswith('e'):
-            framework_route_criteria = is_function
-        elif args.adaptor.lower().startswith('p'):
-            framework_route_criteria = is_function_without_leading_
-        elif args.adaptor.lower().startswith('d'):
-            framework_route_criteria = is_django_view_function
-    # Add all the route functions to the cfg_list
-    FrameworkAdaptor(
-        cfg_list,
-        project_modules,
-        local_modules,
-        framework_route_criteria
-    )
+        framework_route_criteria = is_flask_route_function
+        if args.adaptor:
+            if args.adaptor.lower().startswith('e'):
+                framework_route_criteria = is_function
+            elif args.adaptor.lower().startswith('p'):
+                framework_route_criteria = is_function_without_leading_
+            elif args.adaptor.lower().startswith('d'):
+                framework_route_criteria = is_django_view_function
+        # Add all the route functions to the cfg_list
+        FrameworkAdaptor(
+            cfg_list,
+            project_modules,
+            local_modules,
+            framework_route_criteria
+        )
 
-    initialize_constraint_table(cfg_list)
-    analyse(cfg_list)
-    vulnerabilities = find_vulnerabilities(
-        cfg_list,
-        ui_mode,
-        args.blackbox_mapping_file,
-        args.trigger_word_file,
-        nosec_lines
-    )
+        initialize_constraint_table(cfg_list)
+        analyse(cfg_list)
+        vulnerabilities.extend(find_vulnerabilities(
```

Comment:
![er](https://user-images.githubusercontent.com/19760132/41593157-c7a7d36a-73c7-11e8-94c8-865616f92f06.png)

Reply:
No, there are no vulnerability in a.py b.py and c.py but it printing from xss.py

Reply:
I didn't figure out the bug yet, gonna look more tomorrow 😁 This is harder than expected to track down

Reply:
ok, i fixed it. its about `vulnerabilities = list()` location